src/common/hooks/useSupportsInterface.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
+import { useContractFunctionHook } from "@govtechsg/ethers-contract-hook";
+import { Contract } from "ethers";
+
+interface Erc165Contract extends Contract {
+  supportsInterface: (interfaceId: []) => Promise<boolean> | undefined;
+}
+
+/**
+ * This hook calls checks if token is an instance of given interfaceId
+ */
+export const useSupportsInterface = (contractInstance: Erc165Contract | undefined, interfaceId: string) => {
+  const [isInterfaceType, setIsInterfaceType] = useState<boolean>();
```

Comment:
humm there's no type error here ah? isn't it initialised as undefined?

Reply:
Hmm, what do you mean? The state automatically set the typing to `boolean |  undefined`. Previously as discussed state will be undefined when loading.

Reply:
oh ok i just looked at the type signature of useState, i thought it had to be useState<boolean|undefined> but looks like the undefined is implicit